src/metaswitch/ellis/remote/homestead.py

```diff
@@ -250,29 +259,48 @@ def _location(httpresponse):
         raise HTTPError(500)
 
 
-def _http_request(url, callback, **kwargs):
+def _http_request(url, callback, overload_retries=3, **kwargs):
     http_client = httpclient.AsyncHTTPClient()
     if 'follow_redirects' not in kwargs:
         kwargs['follow_redirects'] = False
     kwargs['allow_ipv6'] = True
-    http_client.fetch(url, callback, **kwargs)
+    retries_holder = {'retries': overload_retries}
+
+    def do_http_request():
+        _log.debug("Sending HTTP request to %s" % url)
+        http_client.fetch(url, callback_wrapper, **kwargs)
+
+    def callback_wrapper(response):
+        _log.debug("Received response from %s with code %d" % (url, response.code))
+        if response.code == 503 and retries_holder['retries'] > 1:
+            _log.debug("503 response - retrying %d more time(s)..." % retries_holder['retries'])
+            retries_holder['retries'] -= 1
+            IOLoop.instance().add_timeout(datetime.timedelta(milliseconds=500), do_http_request)
```

Comment:
I think this makes another request 0.5 seconds later, rather than calling `do_http_request` and making it time out after 0.5 seconds, but I had to think about it - can you add a comment?